### ISSUE TITLE
Close buttons on nbdiff/merge-web page

### ISSUE THREAD

**Issue by martinal** [jupyter/nbdime#111]
Running `nbdiff-web` or `nbmerge-web` on some files now require ctrl+c to shutdown the server, would be nicer with at least a close button like the mergetool view has.

**Comment by vidartf** [comment]
FYI: The server should currently shut down when navigating away from the page (e.g. when closing the tab) via `window.onbeforeunload`. Unfortunately, this will leave the server hanging if the browser crashes (we could have a timeout?) and will also shut down the server if the user refreshes the site.

**Comment by vidartf** [comment]
An alternative solution would be for `nbdiff-web` and `nbmerge-web` to be fire-and-forget servers, in that they automatically shut down the server after the initial API call has been served.

**Comment by martinal** [comment]
It doesn't close for me.

**Comment by martinal** [comment]
Here's why: the close event is only attached if the button has been added:
```
	    // If launched as a tool, there should be a close button, to indicate that
	    // the tool has finshed. If present, wire it to events, and connect to
	    // window unload event as well:
	    var closeBtn = document.getElementById('nbdime-close');
	    if (closeBtn) {
	        closeBtn.onclick = onclose;
	        window.onbeforeunload = onclose;
	    }
```

**Comment by vidartf** [comment]
Ah, you are of course correct. But it is at least possible to enable that behavior on the other tools as well, if we want to.

**Comment by martinal** [comment]
Just adding the buttom to the html templates results in this when clicking it:
```
WARNING:tornado.general:400 POST /api/closetool (127.0.0.1): Server is not a tool server, cannot be closed remotely.
```
Looking at the code, I'm not sure what the right fix is?

**Comment by vidartf** [comment]
The intention there is to prevent remote shut down of normal servers. The current check is rather simplistic; a more extensible way is maybe to add a `closable` flag to the `main_server` function, like this:
```python
def main_server(on_port=None, closable=False, **params):
    params.update({closable: closable})
    app = make_app(**params)

class ApiCloseHandler(NbdimeApiHandler):
    def post(self):
        # Only allow closing, if started as tool
        if not self.params.closable:
            raise web.HTTPError(400, "Server cannot be closed remotely.")
```

**Comment by vidartf** [comment]
Suggested fix along the pattern outlined above in #115.
